pkg/reconciler/integrationsource/resources/containersource.go

```diff
@@ -0,0 +1,231 @@
+/*
+  Copyright 2024 The Knative Authors
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+  	http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+
+package resources
+
+import (
+	"fmt"
+	"reflect"
+	"strconv"
+	"strings"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	sourcesv1 "knative.dev/eventing/pkg/apis/sources/v1"
+	"knative.dev/eventing/pkg/apis/sources/v1alpha1"
+	"knative.dev/pkg/kmeta"
+)
+
+func NewContainerSource(source *v1alpha1.IntegrationSource) *sourcesv1.ContainerSource {
+	return &sourcesv1.ContainerSource{
+		ObjectMeta: metav1.ObjectMeta{
+			OwnerReferences: []metav1.OwnerReference{
+				*kmeta.NewControllerRef(source),
+			},
+			Name:      source.Name,
+			Namespace: source.Namespace,
+		},
+		Spec: sourcesv1.ContainerSourceSpec{
+
+			Template: corev1.PodTemplateSpec{
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						{
+							Name:            "source",
+							Image:           selectImage(source),
+							ImagePullPolicy: corev1.PullIfNotPresent,
+							Env:             makeEnv(source),
+						},
+					},
+				},
+			},
+			SourceSpec: source.Spec.SourceSpec,
+		},
+	}
+}
+
+func generateEnvVarsFromStruct(prefix string, s interface{}) []corev1.EnvVar {
```

Comment:
Some tests for this function would be great

Reply:
yep